inst/include/mode/r/mode.hpp

```diff
@@ -24,10 +26,25 @@ double mode_time(SEXP ptr) {
   return obj->time();
 }
 
+cpp11::sexp state_array(const std::vector<double>& dat,
+                        size_t n_state,
+                        size_t n_particles) {
+  cpp11::writable::doubles ret(dat.size());
+  std::copy(dat.begin(), dat.end(), REAL(ret));
+
+  ret.attr("dim") = cpp11::writable::integers{n_state, n_particles};
```

Comment:
Looks like the compiler is complaining (only on MacOS) that these `size_t` vars can't be cast to integers. What is the best way around this? Cast them explicitly? Change the type of this object?

Reply:
sorry, forgot that this was not well defined. We need to cast them explicitly - you can do this by

```
cpp11::writable::integers{static_cast<int>(n_state), static_cast<int>(n_particles)};
```